### ISSUE TITLE
TSV read/write

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#4746]
At the time this issue is being opened, Zed it at commit b8e1b18.

A user asked in a [community Slack thread](https://brimdata.slack.com/archives/CTSMAK6G7/p1691681741869439):

> I would like for the `zq` command support the TSV-format (Tab-Separated-Values).
> This would make it possible to pipe the output from more programs directly to `zq` for post-processing. For instance if I select a range of cells in Excel and paste it into the terminal I get TSV-data from Excel. Processed with something like `pbpaste | zq -Z -` on a Mac.  Another program that I often use is [VisiData](https://www.visidata.org/) that also outputs TSV by default in pipelines.

We reviewed this one as a team and there was agreement that, particularly given the prevalence of these Excel (and Google Sheets) use cases, it would indeed be useful for us to have direct support for this, e.g.:

1. Explicit `-i tsv` for TSV input
2. Supporting TSV via auto-detect, if feasible
3. `-f tsv` for TSV output

While waiting for this to be implemented, it should be noted that there is already TSV support on the input side, though it's not the easiest to use. The `-csv.delim` option added in #4325 allows the specification of any delimiter when reading via `-i csv`. Unfortunately, it can be kind of a pain to get the tab character past the shell. For instance, given this input data [wine.tsv.gz](https://github.com/brimdata/zed/files/12315989/wine.tsv.gz):

```
$ gzcat wine.tsv.gz 
#	Year	Winery
1	2019	Brutocao
1	2021	Cabana
1	2020	Castoro
```

With my Bash shell, after I type the first `"`, I hit Ctrl-v before hitting Tab and then type the second `"`, and this gets the tab through to `zq` (though unfortunately this doesn't survive cut & paste into & out of the GitHub Issue, so users will need to manually complete the steps in their own shells to see it working).

```
$ zq -version
Version: v1.9.0-7-gb8e1b188

$ zq -i csv -csv.delim "	" wine.tsv.gz 
{"#":1.,Year:2019.,Winery:"Brutocao"}
{"#":1.,Year:2021.,Winery:"Cabana"}
{"#":1.,Year:2020.,Winery:"Castoro"}
```

**Comment by philrz** [comment]
Another community user independently asked for this functionality in a recent [Slack thread](https://brimdata.slack.com/archives/CTT24051C/p1695144673134299).

**Comment by philrz** [comment]
Verified in Zed commit fa48157.

In the attached video I walk through the original community use case of copying some spreadsheet data from Google Sheets that lands as TSV in my paste buffer, then read it successfully with `zq` and turn it into ZSON. I then output the ZSON as TSV back into my paste buffer and paste that into a new spreadsheet on Google Sheets.

https://github.com/brimdata/zed/assets/5934157/2c4098e6-98f0-4344-8d4b-dbb5d618b0e4

While verifying with some other test data, I did find a couple corner cases that need fixing that are described in #4911 and https://github.com/brimdata/zed/pull/4912#issuecomment-1832895486. In the meantime, it's working well enough that I'm going to pursue getting its supported added to Zui (https://github.com/brimdata/zui/issues/2908).

Thanks @mattnibs!
